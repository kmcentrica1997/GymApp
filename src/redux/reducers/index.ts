import {combineReducers} from 'redux';
import usersReducer from './users';

export default combineReducers({
  users: usersReducer,
});

export type RootState = ReturnType<typeof combineReducers>;
