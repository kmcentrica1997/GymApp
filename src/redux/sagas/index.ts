import {all} from '@redux-saga/core/effects';
import watchGetUsersRequest from './users';

export default function* rootSaga() {
  yield all([watchGetUsersRequest()]);
}
