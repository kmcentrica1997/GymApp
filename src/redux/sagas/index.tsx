import todoSagas from './todos';
import {all} from '@redux-saga/core/effects';

export default function* rootSaga() {
  yield all([...todoSagas]);
}
