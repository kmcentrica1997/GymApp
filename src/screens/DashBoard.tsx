import React, {FC, useEffect, useState} from 'react';
import {FlatList, Text, SafeAreaView, StyleSheet, View} from 'react-native';
import axios from 'axios';

const DashBoard: FC = () => {
  return (
    <View>
      <Text>hi</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
export default DashBoard;
