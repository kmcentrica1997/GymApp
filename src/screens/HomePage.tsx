import React, {FC} from 'react';
import {View, StyleSheet} from 'react-native';

const Home: FC = props => {
  return <View />;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
export default Home;
