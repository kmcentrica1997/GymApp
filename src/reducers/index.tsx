import {combineReducers} from 'redux';
import TodosReducer from './todos';

export default combineReducers({
  todos: TodosReducer,
});
