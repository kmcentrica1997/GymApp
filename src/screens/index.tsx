import Login from './Login';
import Home from './HomePage';
import DashBoard from './DashBoard';

export {Login, Home, DashBoard};
