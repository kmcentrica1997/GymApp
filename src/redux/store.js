import createSagaMiddleware from 'redux-saga';
import {createStore, applyMiddleware} from 'redux';
import reducers from './reducers/index';
import rootSaga from './sagas';
import {logger} from './sagas/logger';
const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducers,
  applyMiddleware(sagaMiddleware, logger),
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
);
sagaMiddleware.run(rootSaga);
export default store;
