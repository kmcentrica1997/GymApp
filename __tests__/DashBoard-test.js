import React from 'react';
import {create, act} from 'react-test-renderer';
import {DashBoard} from '../src/screens';
import {createStore, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../src/redux/reducers/index';
import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  {items: 'users'},
  applyMiddleware(sagaMiddleware),
);

const tree = create(
  <Provider store={store}>
    <DashBoard />
  </Provider>,
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
test('item stored properly', () => {
  expect(store.getState().status).toEqual('nothing');
});
