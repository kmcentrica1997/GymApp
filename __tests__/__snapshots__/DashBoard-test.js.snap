// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginTop": 300,
          "textAlign": "center",
          "width": 200,
        }
      }
    >
      Loading...
       
      <Image
        source={
          Object {
            "testUri": "../../../src/img/tired.png",
          }
        }
        style={
          Object {
            "height": 50,
            "marginLeft": 5,
            "width": 50,
          }
        }
      />
    </Text>
  </View>
</View>
`;
